lib/symboldatabase.cpp

```diff
@@ -3987,8 +3987,7 @@ const Function* Scope::findFunction(const Token *tok, bool requireConst) const
                 // check if isConst mismatches
                 if (scope->function->isConst() != func->isConst()) {
                     if (scope->function->isConst()) {
-                        if (!erased)
-                            ++i;
+                        ++i;
```

Comment:
A better fix would be to also move `erased` declaration to line 3972 - right before the foor-loop where is is being changed.